gen/field.go

```diff
@@ -97,11 +115,23 @@ func (f *fieldGroupGenerator) declFieldName(fs *compile.FieldSpec) (string, erro
 	if err != nil {
 		return "", err
 	}
-	if err := f.Reserve(name); err != nil {
-		if fromAnnotation {
-			return "", fmt.Errorf("could not declare field %q (from go.name annotation): %v", name, err)
+
+	if err = f.checkReservedIdentifier(name); err == nil {
+		err = f.Reserve(name)
+	}
+
+	if err != nil {
+		originalName := (name == fs.ThriftName())
+		var note string
+		switch {
+		case originalName && fromAnnotation:
+			note = "(from go.name annotation)"
+		case !originalName && fromAnnotation:
+			note = fmt.Sprintf(" (from %q go.name annotation): %v", fs.ThriftName())
```

Comment:
inconsistent spacing: there will be a space before the `(` in the other two cases but not the first one.